tests/test_examples.py

```diff
@@ -86,6 +100,99 @@ def test_examples_files(test_file: str, symbols_by_file: Dict[str, Set[str]]) ->
     assert found_pylint_message, f"Failed {test_file}. File does not add expected message."
 
 
+def _symbols_by_file_pyta(paths: List[str], include_msg: bool = False) -> Dict[str, Set[str]]:
+    """
+    Run python_ta on all the example files and return the map of file name to the
+    set of PythonTA messages it raises.
+
+    :param paths: The paths to retrieve the files from.
+    :param include_msg: whether to include msgs[msg] in the symbol set
+    """
+    sys.stdout = StringIO()
+    python_ta.check_all(
+        module_name=get_file_paths(paths),
+        config={
+            "output-format": "python_ta.reporters.JSONReporter",
+        }
+    )
+
+    jsons_output = sys.stdout.getvalue()
+    sys.stdout = sys.__stdout__
+    pyta_list_output = json.loads(jsons_output)
+
+    file_to_symbol = {}
+    for path, group in itertools.groupby(pyta_list_output, key=lambda d: os.path.basename(d["filename"])):
+        symbols = set()
+        for message in group:
+            for msg in message["msgs"]:
+                symbols.add(msg["symbol"])
+                if include_msg:
+                    symbols.add(msg["msg"])
+
+        file = os.path.basename(path)
+        file_to_symbol[file] = symbols
+
+    return file_to_symbol
+
+
+@pytest.fixture(scope="session")
+def check_examples_pyta() -> Dict[str, Set[str]]:
+    """
+    Checking the examples files with python_ta
```

Comment:
This description should be improved. This function defines a *fixture* (https://docs.pytest.org/en/7.1.x/explanation/fixtures.html#what-fixtures-are), and so an appropriate description is to describe exactly what the fixture is. Similarly, the name `check_examples_pyta` isn't a good name; pick a *noun phrase* for the fixture.

Same comments apply to `check_pycodestyle_pyta` below.